lib/runtime/src/ops/layer_norm.cc

```diff
@@ -16,566 +16,222 @@
+  auto input = acc.get_tensor<Permission::RO>(INPUT);
+  FFHandler handle = acc.get_argument<FFHandler>(HANDLE);
 
-/*
-  regions[0](I): output_grad
-  regions[1](I): input
-  regions[2](I/O): input_grad
-  regions[3](I): gamma
-  regions[4](I/O): gamma_grad
-  regions[5](I/O): beta_grad
-   */
-void LayerNorm::backward_task(Task const *task,
-                              std::vector<PhysicalRegion> const &regions,
-                              Context ctx,
-                              Runtime *runtime) {
-  LayerNormMeta const *m = *((LayerNormMeta **)task->local_args);
-  assert(task->regions.size() == regions.size());
-  float const *in_ptr = NULL, *out_grad_ptr = NULL, *gamma_ptr = NULL;
-  float *in_grad_ptr = NULL, *gamma_grad_ptr = NULL, *beta_grad_ptr = NULL;
-  Domain out_grad_domain = runtime->get_index_space_domain(
-      ctx, task->regions[0].region.get_index_space());
-  out_grad_ptr = helperGetTensorPointerRO<float>(
-      regions[0], task->regions[0], FID_DATA, ctx, runtime);
-  Domain in_domain = runtime->get_index_space_domain(
-      ctx, task->regions[1].region.get_index_space());
-  in_ptr = helperGetTensorPointerRO<float>(
-      regions[1], task->regions[1], FID_DATA, ctx, runtime);
-  Domain in_grad_domain = runtime->get_index_space_domain(
-      ctx, task->regions[2].region.get_index_space());
-  in_grad_ptr = helperGetTensorPointerRW<float>(
-      regions[2], task->regions[2], FID_DATA, ctx, runtime);
-  assert(in_domain == out_grad_domain);
-  assert(in_domain.get_volume() ==
-         m->effective_num_elements * m->effective_batch_size);
-  if (m->elementwise_affine) {
-    assert(regions.size() == 6);
-    Domain gamma_domain = runtime->get_index_space_domain(
-        ctx, task->regions[3].region.get_index_space());
-    gamma_ptr = helperGetTensorPointerRO<float>(
-        regions[3], task->regions[3], FID_DATA, ctx, runtime);
-    Domain gamma_grad_domain = runtime->get_index_space_domain(
-        ctx, task->regions[4].region.get_index_space());
-    gamma_grad_ptr = helperGetTensorPointerRW<float>(
-        regions[4], task->regions[4], FID_DATA, ctx, runtime);
-    Domain beta_grad_domain = runtime->get_index_space_domain(
-        ctx, task->regions[5].region.get_index_space());
-    beta_grad_ptr = helperGetTensorPointerRW<float>(
-        regions[5], task->regions[5], FID_DATA, ctx, runtime);
-    assert(gamma_domain == gamma_grad_domain);
-    assert(gamma_domain == beta_grad_domain);
-    assert(gamma_domain.get_volume() == m->effective_num_elements);
-  } else {
-    assert(regions.size() == 3);
+  // question: how to get batch_size and effective_num_elements
+  int64_t effective_batch_size, effective_num_elements;
+  int M = 1;
+  for (int i = 0; i < attrs.axes.size(); i++) {
+    M *= input.shape.at(legion_dim_t(attrs.axes[i]));
+  }
+  int num_replicas = 1;
+  for (int i = 0; i < intput.shape.num_dims(); i++) {
+    num_replicas *= input.shape.at(legion_dim_t(i));
```

Comment:
In the original code I don't think it's divided by `num_replicas` to get `effective_batch_size`. Just `inputs->volume() / M`